include/avif/avif.h

```diff
@@ -628,9 +604,35 @@ AVIF_API uint32_t avifRGBImagePixelSize(const avifRGBImage * rgb);
 AVIF_API void avifRGBImageAllocatePixels(avifRGBImage * rgb);
 AVIF_API void avifRGBImageFreePixels(avifRGBImage * rgb);
 
-// The main conversion functions
-AVIF_API avifResult avifImageRGBToYUV(avifImage * image, const avifRGBImage * rgb);
-AVIF_API avifResult avifImageYUVToRGB(const avifImage * image, avifRGBImage * rgb);
+// Conversion options.
+typedef enum avifConversionFlag
+{
+    AVIF_CONVERSION_DEFAULT = 0, // Default settings.
+                                 // Uses the first available upsampling filter among:
+                                 //   libyuv average, built-in average
+                                 // Uses the first available downsampling filter among:
+                                 //   libyuv bilinear, libyuv nearest-neighbor, built-in bilinear
+
+    // libyuv preference
+
+    AVIF_CONVERSION_AVOID_LIBYUV = (1 << 0), // Only picks built-in paths. The default is to prefer libyuv if available.
+
+    // Conversion from RGB to YUV 4:2:2 or YUV 4:2:0 (ignored in all other cases)
+
+    // Chroma downsampling filter. Set at most one:
+    AVIF_CHROMA_DOWNSAMPLING_AVERAGE = (1 << 10), // only use the averaging filter (libyuv or built-in)
+
+    // Conversion from YUV 4:2:2 or YUV 4:2:0 to RGB (ignored in all other cases)
+
+    // Chroma upsampling filter. Set at most one:
+    AVIF_CHROMA_UPSAMPLING_NEAREST = (1 << 20), // only use the fast nearest-neighbor filter (libyuv or built-in)
+    AVIF_CHROMA_UPSAMPLING_BILINEAR = (1 << 21) // only use the good-quality bilinear filter (libyuv or built-in)
```

Comment:
> 1. Since chroma downsampling and chroma upsampling flags are never used at the same time, they do not need to be assigned to different bit positions.

Based on this and the [other](https://github.com/AOMediaCodec/libavif/pull/1026#discussion_r954314330) comments, I decided to split `avifConversionFlags` into `avifRGBToYUVFlags` and `avifYUVToRGBFlags`.

> 2. One way to ensure at most one chroma upsampling filter is set is to number them `1 << 20`, `2 << 20`, `3 << 20`, etc., instead of `1 << 20`, `1 << 21`, `1 << 22`.

I would rather not mix bit flags and enum constants in the same enum. `(2 << 20) == (1 << 21)` so the values are the same for two possibilities anyway.